src/backend/src/services/announcement.service.ts

```diff
@@ -45,6 +45,66 @@ export default class AnnouncementService {
     return announcementTransformer(announcement);
   }
 
+  static async updateAnnouncement(
+    text: string,
+    usersReceivedIds: string[],
+    dateMessageSent: Date,
+    senderName: string,
+    slackEventId: string,
+    slackChannelName: string,
+    organizationId: string
+  ): Promise<Announcement> {
+    const originalAnnouncement = await prisma.announcement.findUnique({
+      where: {
+        slackEventId
+      }
+    });
+
+    if (!originalAnnouncement) throw new NotFoundException('Announcement', slackEventId);
+
+    const announcement = await prisma.announcement.update({
+      where: { announcementId: originalAnnouncement.announcementId },
+      data: {
+        text,
+        usersReceived: {
+          set: usersReceivedIds.map((id) => ({
+            userId: id
+          }))
+        },
+        slackEventId,
+        dateMessageSent,
+        senderName,
+        slackChannelName
+      },
+      ...getAnnouncementQueryArgs(organizationId)
+    });
+
+    return announcementTransformer(announcement);
+  }
+
+  static async deleteAnnouncement(slackEventId: string, organizationId: string): Promise<Announcement> {
+    const originalAnnouncement = await prisma.announcement.findUnique({
+      where: {
+        slackEventId
+      }
+    });
+
+    if (!originalAnnouncement) throw new NotFoundException('Announcement', slackEventId);
```

Comment:
check if its already deleted

Reply:
check if its not in this organization

Reply:
announcements don't have organizations, so im not sure what this means